src/rgw/rgw_op.cc

```diff
@@ -1103,27 +1103,71 @@ int RGWGetObj::verify_permission(optional_yield y)
     if (has_s3_existing_tag || has_s3_resource_tag)
       rgw_iam_add_objtags(this, s, has_s3_existing_tag, has_s3_resource_tag);
 
-  if (get_torrent) {
-    if (s->object->get_instance().empty()) {
-      action = rgw::IAM::s3GetObjectTorrent;
-    } else {
-      action = rgw::IAM::s3GetObjectVersionTorrent;
+  // if this is a system request and the user is not an admin of the bucket, then it's an impersonation of another user
+  // so we need to check for perms
+  bool is_replication_request = s->system_request && !s->auth.identity->is_admin();
```

Comment:
not sure the `is_admin()` part is necessary here, given that we're in `verify_permission()` and will override permission errors if we are. isn't `s->system_request` enough to opt into the multisite extension?